File: mobile/lib/models/lens_option.dart
class LensOption {
  final String id;
  final String kind; // 'single_vision', 'progressive', etc.
  final String? lensType;
  final String? subType;
  final String? qualityTier;
  final String name;
  final double price;
  final List<String> features;
  final bool isRecommended;
  final String? description;
  final String? badge;

  LensOption({
    required this.id,
    required this.kind,
    this.lensType,
    this.subType,
    this.qualityTier,
    required this.name,
    required this.price,
    this.features = const [],
    this.isRecommended = false,
    this.description,
    this.badge,
  });

  factory LensOption.fromJson(Map<String, dynamic> json) => LensOption(
        id: json['_id'] ?? json['id'] ?? '',
        kind: json['type'] ?? '',
        lensType: json['type'],
        subType: json['subType'],
        qualityTier: json['subType'],
        name: json['displayName'] ?? json['name'] ?? '',
        price: (json['price'] as num?)?.toDouble() ?? 0,
        features: List<String>.from(json['features'] ?? []),
        isRecommended: json['badge'] == 'RECOMMENDED',
        description: json['description'],
        badge: json['badge'],
      );
}

File: web/src/models/LensOption.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface ILensOption extends Document {
  kind: 'type' | 'quality';
  type?: string;
  subType?: string;
  displayName: string;
  name: string;
  description?: string;
  price: number;
  startingPrice?: number;
  features: string[];
  badge?: string;
  isBestseller: boolean;
  isRecommended: boolean;
  sortOrder: number;
  isActive: boolean;
}

const LensOptionSchema = new Schema<ILensOption>({
  kind: { type: String, enum: ['type', 'quality'], required: true },
  type: {
    type: String,
    enum: ['single_vision', 'progressive', 'bluecut', 'zero_power', 'photochromic'],
  },
  subType: { type: String },
  displayName: { type: String, required: true },
  name: { type: String, required: true },
  description: { type: String },
  price: { type: Number, required: true },
  startingPrice: { type: Number },
  features: [String],
  badge: { type: String },
  isBestseller: { type: Boolean, default: false },
  isRecommended: { type: Boolean, default: false },
  sortOrder: { type: Number, default: 0 },
  isActive: { type: Boolean, default: true },
});

export const LensOption = mongoose.models.LensOption || mongoose.model<ILensOption>('LensOption', LensOptionSchema);
